lib/phoenix/router/console_formatter.ex

```diff
@@ -10,14 +10,36 @@ defmodule Phoenix.Router.ConsoleFormatter do
   @longpoll_verbs ["GET", "POST"]
 
   def format(router, endpoint \\ nil) do
-    routes = Phoenix.Router.routes(router)
+    routes = 
+      router
+      |> Phoenix.Router.routes()
+      |> expand_dynamic_routes()
+
     column_widths = calculate_column_widths(router, routes, endpoint)
 
     routes
     |> Enum.map_join("", &format_route(&1, router, column_widths))
     |> Kernel.<>(format_endpoint(endpoint, router, column_widths))
   end
 
+  defp expand_dynamic_routes(routes) do
+    Enum.flat_map(routes, fn route -> 
+      if function_exported?(route.plug, :phoenix_routes, 1) do
```

Comment:
I believe Code.ensure_loaded? will be required here.

Reply:
I had thought so as well but it worked without it. No problem adding it back in.